call it plugin without Extbase (#4458)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/FlexForms/Index.rst

        );

>     If you are using a content element plugin without Extbase, the example

Reviewer: ```suggestion
    If you are using a content element instead (not a "General Plugin" which has :sql:`CType` set to `list_type`), the example
```

Author: See my comment below.

Reviewer 2: I think that makes it quite clear. This is not about Extbase / non-Extbase but about cType=list vs cType=tx_my_custom_type

Author: Please ask Stefan Bürk. The `cType=list` should not be used any more as he has explained on Slack.

Reviewer 2: It is not deprecated yet. I think @sbuerk was expressing his opinion and not official policy.

Author: Let's wait for @sbuerk . He knows more than we do.

Reviewer 3: it's not officially deprecated yet, but it's a official task to remove the subtype handling from the tt_content (CType (maintype) = list, list_type (subtype) = ...).

Part of that have been to move internal subtype plugins to maintype plugins (e.g. Frontend Login, indexed-search plugin, ...) - not sure what's missing yet, and when the deprecation really happens. But it's on a shared/global to-do list and not a personal optionion.

Literally, there have never been any benifit of doing plugins or extbase plugins as subtype (CType = list, list_type = ...) and it was always possible to register them as CTypes.

Basically, we are talking of content-element in general, which can be simple content objects of different implementation (text, textmedia, ...), plugins, extbase plugins, ... technically, they simply need a rendering method, and here the differents came into play.

On top of that, there was the "General Plugin" thingy which was based on the "subtype" feature, handled by the `cType = list` content element - and having the dedicated select box for "plugins". I guess, from that case the thinking of `General Plugins` originate from that. In the end, it's a plugin/extbase plugin - just handled as subtype instead of a (main)type aka content element type.

Note: Literally, it would have been possible to do "content element" stuff on subtype level also, using direct manipulation of the global arrays and other places.

I do not have any suggestion of wording or how to explain that in different places and contexts.

I recommended to @franzholz to switch to CType for the plugin registration directly due to his slowly adoption rate for new technologies - and having to change something (cannot remember if it was related to switchable controller removal or other stuff it would be better to do that directly as CType to be on the safe side for upcoming versions, where at some point the deprecation of the subtype feature will happen and be removed at some point. Not sure if we depracte it in v13 or v14, but it will happen.

So sad, it may be usefull to have something explaining the difference / type + subtype feature in some way, and pointing on the differences for the one or other stuff, like registering calls, adding PI flexforms (which must be in line with the choosen type/subtype) and other stuff.

So to say `General Plugin` may be known as the `subtype` feature.

On top of that, it was (and is still) possible to add a custom CType (main/content-element type) which introduces a custom sub-type handling - albeit being quite some effort, but possible (and the reason noone has done that).

From a technical point of view, removing the subtype feature ( `list_type` ) reduces a lot of headache and handling in a lot of places. And from a user point of view, it's better to have them available directly in as content element.

For example, with the new site sets stuff it is possible to control which content elements are allowed on a site root, and `ext:styleguide` removes the general plugin from the new content element already (having it still available as ctype in the normal edit).

>     If you are using a content element with a custom CType (recommended), the example

Reviewer 2: ```suggestion
    If you are using a content element with a custom CType (recommend, both with and without 
    Extbase), the example
```

>     If you are using a content element instead (not a "General Plugin" which has :sql:`CType` set to `list` and uses :sql:`list_type` to be configured), the example

Reviewer 2: ```suggestion
    If you are using a content element with a custom CType (recommended), the example
```
